web/auth/auth.go

```diff
@@ -47,28 +51,25 @@ func newUserSession(id uint64) *UserSession {
 	}
 }
 
-// UserToken holds a map from session cookies to user IDs.
-type UserToken struct {
-	store map[string]uint64
+func (us *UserSession) enableProvider(provider string) {
+	us.Providers[provider] = struct{}{}
 }
 
-func (ut *UserToken) add(token string, id uint64) {
-	for token, userID := range TokenStore.store {
-		if userID == id {
-			delete(TokenStore.store, token)
+// map from session cookies to user IDs.
+var tokenStore = make(map[string]uint64)
```

Comment:
Should we use the term `cookie` instead of `token` for this stuff? (including the `Add(token ...` method below)

Reply:
Agreed, that would make more sense.